src/schema/mod.rs

```diff
@@ -114,11 +114,17 @@ impl FederationSchema {
     ) -> Result<IndexSet<ObjectTypeDefinitionPosition>, FederationError> {
         Ok(match composite_type_definition_position {
             CompositeTypeDefinitionPosition::Object(pos) => IndexSet::from([pos]),
-            CompositeTypeDefinitionPosition::Interface(pos) => self
-                .referencers()
-                .get_interface_type(&pos.type_name)?
-                .object_types
-                .clone(),
+            CompositeTypeDefinitionPosition::Interface(pos) => {
+                let interface_type = self.referencers().get_interface_type(&pos.type_name)?;
+                let mut possible_runtimes = interface_type.object_types.clone();
+                for i in interface_type.interface_types.clone() {
+                    self.possible_runtime_types(CompositeTypeDefinitionPosition::Interface(
+                        i.clone(),
+                    ))
+                    .map(|r| possible_runtimes.extend(r))?;
+                }
```

Comment:
Is this necessary?

This would look up transitive interfaces, but you always have to write out transitive interfaces in the `implements` clause, so any object type implementing a transitive interface should already be in the `referencers()` of the base interface.

Reply:
hmmm unsure? I could have sworn that I added this to fix a unit test... but I just tried with the original logic and everything still works 🤷 i'll revert this change